### DISCUSSION TITLE
ERROR   -  Config value 'plugins': The "minify" plugin is not installed

### DISCUSSION THREAD

**Discussion by kuhlaid** [squidfunk/mkdocs-material#7485] [Q&A] [answered]
### Bug description

I forked the latest version mkdocs-material-9.5.33 and tried to get it pushed to GitHub Pages. After creating a workflow action in GitHub as outlined in https://squidfunk.github.io/mkdocs-material/publishing-your-site/ the action failed with `ERROR   -  Config value 'plugins': The "minify" plugin is not installed`. I have not altered the code nor performed any other actions outside of GitHub.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

NA

### Steps to reproduce

1. Fork the latest version mkdocs-material-9.5.33
2. Create a workflow action in GitHub as outlined in https://squidfunk.github.io/mkdocs-material/publishing-your-site/
3. Action fails with `ERROR   -  Config value 'plugins': The "minify" plugin is not installed`.

### Browser

Edge

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Moving to discussions, as this is clearly a support request.

**Comment by kuhlaid** [comment]
This seems like a bug in the configuration. I do not see any mention of minify in the list of built-in plugins under https://squidfunk.github.io/mkdocs-material/plugins/?h=plugin so where is the minify plugin coming from if it is not built-in?

**Reply by kamilkrzyskow** [reply, marked as answer]
Hello @kuhlaid,
the workflow in the publishing guide installs the theme, builds mkdocs together with the deployment:

```
- run: pip install mkdocs-material 
- run: mkdocs gh-deploy --force
```

> ERROR   -  Config value 'plugins': The "minify" plugin is not installed

This is caused due to your `mkdocs.yml` file containing the `minify` plugin, but you didn't install it separately.

You can do so by installing the extra recommended packages:
Replace this `pip install mkdocs-material ` with `pip install mkdocs-material[recommended]`.

Another approach is also to add one more step in the workflow before mkdocs is built:
Add a `run: pip install mkdocs-minify-plugin` step.

BTW I'm curious how there is the minify plugin in your `mkdocs.yml` file and you don't know where it came from? Where did you get the mkdocs.yml file from?

EDIT: There is also the other plugin:
- https://github.com/monosans/mkdocs-minify-html-plugin
but it has a different API than the more common `mkdocs-minify-plugin`

> [!NOTE]
> GitHub is bugged recently and I get no notifications for discussions converted from Issues 🤔, I found this by chance

**Reply by kamilkrzyskow** [reply]
@kuhlaid 
Have you resolved the issue?

**Reply by kuhlaid** [reply]
Like I said, I did not create nor change any of the code. I simply forked the repo 'as is' and tried to get GitHub pages to build the site.

**Reply by kamilkrzyskow** [reply]
> Like I said, I did not create nor change any of the code

Not true, you did create a custom workflow file based on the [publishing guide](https://squidfunk.github.io/mkdocs-material/publishing-your-site/), and this expects you to follow the previous [Installation](https://squidfunk.github.io/mkdocs-material/getting-started/) and [Creating your site](https://squidfunk.github.io/mkdocs-material/creating-your-site/) instructions, to create an example bare MkDocs project `mkdocs new .` with the Material theme enabled. The `minify` plugin isn't enabled there, so it's not added to the publishing guide.

The repository in itself uses a custom repository-specific `mkdocs.yml` file, which enables the `minify` plugin and requires different processes enabled that are present in the `documentation.yml` workflow file, like I mentioned in my comment above.
In general there is a lot of custom stuff added to the workflow, so better don't use it as is, just take out the installation of additional dependencies and include it in your custom workflow based on the publishing guide.


- https://github.com/squidfunk/mkdocs-material/blob/d95e0e62def5068ec8003d378c164ab5b752e21c/mkdocs.yml#L99-L100
- https://github.com/squidfunk/mkdocs-material/blob/d95e0e62def5068ec8003d378c164ab5b752e21c/.github/workflows/documentation.yml#L71-L74

**Reply by kamilkrzyskow** [reply]
You could also follow along with the https://squidfunk.github.io/mkdocs-material/tutorials/ as there are templates you should be able to use `as-is`, while following along with the tutorial.

**Reply by kuhlaid** [reply]
I changed the workflow to use `pip install mkdocs-material[recommended]` and that worked. Thank you. I would suggest using this in the workflow documentation or some other workflow that is compatible with the framework out of the box because this was not an obvious fix to the issue.

**Reply by kamilkrzyskow** [reply]
OK, thanks for the feedback, the guides might be changed/reassessed in the future. 

For now my point stands, that the guides on the documentation page, expect following instructions in order `Installation -> Creating your site -> Publishing guide`, and creating a new bare project. not reusing the repository files, while using guides from the "Getting Started" section.

The usage of repository files is typically reserved for modifying the project itself, not building something new out of them.
I feel like this "approach" is common across different GitHub projects that I've seen. The Customization guide does explain that for theme development the `recommended` packages should be installed:
- https://squidfunk.github.io/mkdocs-material/customization/#theme-development
